src/Compilers/Core/Portable/NativePdbWriter/PdbWriter.cs

```diff
@@ -804,6 +811,56 @@ private static object CreateSymWriterWorker()
             return symWriter;
         }
 
+        [DllImport("kernel32")]
+        private static extern IntPtr LoadLibrary(string path);
+
+        [DllImport("kernel32")]
+        private static extern bool FreeLibrary(IntPtr hModule);
+
+        [DllImport("kernel32")]
+        private static extern IntPtr GetProcAddress(IntPtr hModule, string procedureName);
+
+        private delegate void CreateSymWriterDelegate(ref Guid id, [MarshalAs(UnmanagedType.IUnknown)]out object symWriter);
+
+        private static object TryLoadFromAlternativePath()
+        {
+            var dir = Environment.GetEnvironmentVariable("MICROSOFT_DIASYMREADER_NATIVE_ALT_LOAD_PATH");
+            if (string.IsNullOrEmpty(dir))
+            {
+                return null;
```

Comment:
>return null; [](start = 16, length = 12)

Should we re-throw the original exception instead? Why is returning null the right thing to do when the variable isn't set.

Reply:
Returning null will cause falling back to the globally registered instance.

Reply:
Here: https://github.com/dotnet/roslyn/pull/21724/files/373bedffa03ba5462c8fa4148f04cbd943812269#diff-52b4c1e8d216bc9d8d3f4978b002f9a9R805

Reply:
I agree returning `null` will fallback to global registered instance. However there is still a deviation from the orginial behavior. Prior to this change it would update s_MicrosoftDiaSymReaderNativeLoadFailure to e.Message and then fall back to global. Now it just falls back. Seems like we should be updating that failure.

Reply:
Not having the environment variable set is not an error. s_MicrosoftDiaSymReaderNativeLoadFailure is only set on error.

Reply:
Sure. But in this case there was an exception that caused us to go to this path (which is an error). This alternate idea failed. While that isn't an error the totality of the operation still had an error and seems like tha should be set. At least it would've been set before this change.

That's how I'm seeing the code at least.

Reply:
Oh, I see. You are right.

Reply:
Fixed and added more tests.